Error on missing "value" in template arguments

When running on the following `WorkflowTemplate`:

```
apiVersion: argoproj.io/v1alpha1
kind: WorkflowTemplate
metadata:
  name: test-pipeline
  namespace: argo-workflows

spec:
  entrypoint: hello-world

  arguments:
    parameters:
      - name: test-argument
  templates:
    - name: hello-world         # name of the template
      container:
        image: busybox
        command: [ echo ]
        args: [ "hello world" ]
        resources: # limit the resources
          limits:
            memory: 32Mi
            cpu: 100m
```

I receive the following error:

```
example.yaml:12:9: M101 Missing required field

 10 |   arguments:
 11 |     parameters:
 12 |       - name: test-argument
    |         └ M101 at .spec.arguments.parameters.0 in test-pipeline
 13 |   templates:
 14 |     - name: hello-world         # name of the template

    Missing required field for the 'parameters' section.
    One of the following fields is required: 'value' or 'valueFrom'.

Found 1 failures
```

But this workflow template is working fine. The value is passed in when creating the `Workflow` object. Is this a bug?

--- a/tugboat/analyzers/workflow.py
+++ b/tugboat/analyzers/workflow.py
@@ -156,17 +156,19 @@ def check_argument_parameters(workflow: WorkflowCompatible) -> Iterator[Diagnosi
             loc=loc,
             fields=["name"],
         )
-        yield from require_exactly_one(
-            model=param,
-            loc=loc,
-            fields=["value", "valueFrom"],
-        )
         yield from accept_none(
             model=param,
             loc=loc,
             fields=["default", "enum"],
         )
         yield from prepend_loc(loc, critique_relaxed_parameter(param))
+
+        if workflow.kind == "Workflow":
+            yield from require_exactly_one(
+                model=param,
+                loc=loc,
+                fields=["value", "valueFrom"],
+            )
 
         if param.valueFrom:
             yield from require_exactly_one(

